### PR TITLE
Add Precision, Recall and F1 Score metrics in callbacks.py

### PR DESCRIPTION
**Pull request by SriRangaTarun**
### Summary
Add precision, recall and f1 score metrics for binary classification.

### Related Issues
[This issue](https://github.com/keras-team/keras/issues/11846)

### PR Overview
Added precision, recall and f1 score metrics for binary classification in callbacks.py.

- [n] This PR requires new unit tests [y/n] (make sure tests are included)
- [y] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by arquolo** [comment]
External dependencies (not in [setup.py](https://github.com/keras-team/keras/blob/master/setup.py)) are forbidden.
You can try to implement metrics via stateful Layer, it will be better than triggering `predict` in each metric and forcing people to install `scikit-learn`.
